### ISSUE TITLE
Categorical ordering, it's not respected for Boxplot and violin plot 

### ISSUE THREAD

**Issue by icedevil2001** [JetBrains/lets-plot#746]
I am trying to order the X axis in chronological order, which has been set as a categorical variable. When I perform a facet plot, I find that the X-axis does not respect the categorical ordering when one or more chromosome arms have missing values for the entire chromosome arm.
I am very new to the Lets-plot library and have never used  ggplot

Below is a minimal example

```python 
import pandas as pd
n = 100
np.random.seed(42)
chrom_order = ['chr1', 'chr2', 'chr4', 'chr5']

chrom = np.random.choice(chrom_order, size=n)
arm = np.random.choice(['p',"q"], size=n)
y = np.random.normal(size=n)

df = pd.DataFrame({'chrom': chrom, 'y': y, 'arm': arm})
df['chrom'] = pd.Categorical(df.chrom, categories=chrom_order, ordered=True)

df.loc[(df.chrom == "chr1")& (df.arm == 'p') , 'y'] = np.NAN
df.dropna(subset='y', inplace=True)
df = df.sort_values("chrom")
p3 = (
    ggplot(df, aes(x="chrom", y='y')) 
    + geom_boxplot()
    + facet_grid(y='arm')
    )

p3.show()
```
<img width="596" alt="image" src="https://user-images.githubusercontent.com/11421945/229010174-4278eb14-0e42-4596-b05b-7f675d6e7ee9.png">

**Comment by alshan** [comment]
Related #679

**Comment by icedevil2001** [comment]
Super!! Thank you for the quick response and  both solutions work excellently!

**Comment by alshan** [comment]
Fixed in [v3.2.0](https://github.com/JetBrains/lets-plot/blob/master/CHANGELOG.md#320---2023-05-09)
